miso-web/src/main/webapp/scripts/admin_ajax.js

```
@@ -0,0 +1,63 @@
+ * (at your option) any later version.
+ *
+ * MISO is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MISO.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * *********************************************************************
+ */
+
+var Admin = Admin || {
+  clearCache : function() {
+    Utils.showConfirmDialog('Hibernate Cache', 'Clear',
+        [ 'Clear Hibernate cache?' ], function() {
+          Utils.ajaxWithDialog('Clearing Cache', 'POST',
+              '/miso/rest/admin/cache/clear', null, function(success) {
+                Utils.showOkDialog('Cache', [ success ? 'Cache cleared.'
+                    : 'Failed to clear cache.' ]);
+              });
+        });
+  },
+  regenBarcodes : function() {
+    Utils
+        .showConfirmDialog(
+            'Barcodes',
+            'Generate',
+            [ 'Generate missing barcodes?' ],
+            function() {
+              Utils
+                  .ajaxWithDialog(
+                      'Clearing Cache',
+                      'POST',
+                      '/miso/rest/admin/barcode/regen',
+                      null,
+                      function(results) {
+                        Utils
+                            .showOkDialog(
+                                'Cache',
+                                results
+                                    .filter(function(result) {
+                                      return result.count > 0;
+                                    })
+                                    .map(
+                                        function(result) {
+                                          return "Regenerated " + result.count + " barcodes for " + result.target + ".";
```

Controller logs and ignores errors, and then this only shows the success count. Unless you know the exact number of objects you expect this to work for (or check the log when you have no reason to), there's no indication that anything failed. Should at least count failures and notify the user

---

Okay. Included some more stats.